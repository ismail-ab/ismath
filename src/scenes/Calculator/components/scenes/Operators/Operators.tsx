import * as React from 'react';
import { CharacterButton } from '../../components';

const Operators: React.SFC = () => {
  const operators: string[] = ['+', '−', '×', '÷'];

  const createOperatorsButton = () => operators.map(operator => (
    <CharacterButton
      key={operator}
      characterToDisplay={operator}
      color="secondary"
    />
  ));

  return (
    <>
      {createOperatorsButton()}
    </>
  );
};

export default Operators;