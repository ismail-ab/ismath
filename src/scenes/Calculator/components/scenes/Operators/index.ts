export { default as Operators } from './Operators';