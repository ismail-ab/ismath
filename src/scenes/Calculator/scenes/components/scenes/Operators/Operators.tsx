import * as React from 'react';
import { CharacterButton } from '../../components';

interface IOperator {
  operatorForCalculate: string;
  operatorToDisplay: string;
}

const Operators: React.SFC = () => {
  const operatorsArray: IOperator[] = [
    {
      operatorForCalculate: '+',
      operatorToDisplay: '+'
    },
    {
      operatorForCalculate: '-',
      operatorToDisplay: '−'
    },
    {
      operatorForCalculate: '*',
      operatorToDisplay: '×'
    },
    {
      operatorForCalculate: '/',
      operatorToDisplay: '÷'
    }
  ];

  const createOperatorsButton = () => operatorsArray.map(operator => (
    <CharacterButton
      key={operator.operatorForCalculate}
      characterForCalculate={operator.operatorForCalculate}
      characterToDisplay={operator.operatorToDisplay}
      color="secondary"
    />
  ));

  return (
    <>
      {createOperatorsButton()}
    </>
  );
};

export default Operators;