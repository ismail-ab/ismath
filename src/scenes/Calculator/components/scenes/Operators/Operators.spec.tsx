import { } from 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';
import Operators from './Operators';

describe('Operators', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Operators />);

    expect(wrapper).toMatchSnapshot();
  });
});
