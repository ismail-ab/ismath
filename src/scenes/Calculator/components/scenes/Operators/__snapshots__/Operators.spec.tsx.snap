// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operators renders correctly 1`] = `
<React.Fragment>
  <CharacterButton
    characterToDisplay="+"
    color="secondary"
    key="+"
  />
  <CharacterButton
    characterToDisplay="−"
    color="secondary"
    key="−"
  />
  <CharacterButton
    characterToDisplay="×"
    color="secondary"
    key="×"
  />
  <CharacterButton
    characterToDisplay="÷"
    color="secondary"
    key="÷"
  />
</React.Fragment>
`;
